### DISCUSSION TITLE
Entity metadata

### DISCUSSION THREAD

**Discussion by azarenkovgd** [mikro-orm/mikro-orm#3848] [Q&A] [answered]
How to get [Entity Metadata](https://mikro-orm.io/api/core/class/EntityMetadata) having `EntityManager` and entity class (for example `SomeEntity`)? I know about `em.getMetadata().get(...)`, but `get` requires string param (table name as I understand). But I might not know table name at a runtime without metadata (maybe there are others way to get table name I do not know about).

For context - I want to iterate over entity relations and do some staff based on this information.

**Comment by B4nan** [comment, marked as answer]
Its not table name, its entity name. So just do `em.getMetadata().get(MyEntity.name)`.

The metadata copy is also available in the entity prototype, so something like `MyEntity.prototype.__meta` should work too (it won't type check though).

**Reply by azarenkovgd** [reply]
Thank you!
